### PR TITLE
Add bit vector support for 8

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds bit vector support for 8 bits.

This is needed to support the upcoming hard fork when prysm implements the new incentive accounting reform proposal. 
It'll be used for `current_epoch_participation` and `prev_epoch_participation` in beacon state:
```
List[Bitvector[PARTICIPATION_FLAGS_LENGTH], VALIDATOR_REGISTRY_LIMIT]
```

Learn more: https://github.com/ethereum/eth2.0-specs/pull/2176

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]

File: bitvector8.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func (b Bitvector8) Len() uint64 {
+	return 8
+}
+
+// Count returns the number of 1s in the bitvector.
+func (b Bitvector8) Count() uint64 {
+	if len(b) == 0 {
+		return 0
+	}
+	return uint64(bits.OnesCount8(b.Bytes()[0]))
```
Comment (prestonvanloon) [inline bitvector8.go:58]:
```suggestion
	return uint64(bits.OnesCount8(b[0])
```
this would be faster like this

File: bitvector8.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+// Bytes returns the bytes data representing the Bitvector8. This method
+// bitmasks the underlying data to ensure that it is an accurate representation.
+func (b Bitvector8) Bytes() []byte {
+	if len(b) == 0 {
+		return []byte{}
+	}
+	return []byte{b[0]}
```
Comment (prestonvanloon) [inline bitvector8.go:67]: Shouldn't this just return `b`?

**Comment by terencechain** [comment]
Thanks @prestonvanloon !

File: bitvector8.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+func (b Bitvector8) Len() uint64 {
+	return 8
+}
+
+// Count returns the number of 1s in the bitvector.
+func (b Bitvector8) Count() uint64 {
+	if len(b) == 0 {
+		return 0
+	}
+	return uint64(bits.OnesCount8(b[0])
```
Comment (prestonvanloon) [inline bitvector8.go:58]:
```suggestion
	return uint64(bits.OnesCount8(b[0]))
```

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**